datalad_next/archive_operations/__init__.py

```diff
@@ -0,0 +1,65 @@
+"""Archive operation handlers"""
+
+# allow for |-type UnionType declarations
+from __future__ import annotations
+
+import logging
+from abc import (
+    ABC,
+    abstractmethod,
+)
+from contextlib import contextmanager
+from typing import (
+    Any,
+    Generator,
+    IO,
+)
+
+import datalad
+
+from datalad_next.config import ConfigManager
+from datalad_next.iter_collections.utils import FileSystemItem
+
+
+lgr = logging.getLogger('datalad.ext.next.archive_operations')
+
+
+class ArchiveOperations(ABC):
+    """
+    """
+    def __init__(self, location: Any, *, cfg: ConfigManager | None = None):
+        """
+        Parameters
+        ----------
+        location:
+          Archive location identifier (path, URL, etc.) understood by a
+          particular archive handler.
+        cfg: ConfigManager, optional
+          A config manager instance that implementations will consult for
+          any configuration items they may support.
+        """
+        self._cfg = cfg
+
+    @property
+    def cfg(self) -> ConfigManager:
+        if self._cfg is None:
+            self._cfg = datalad.cfg
+        return self._cfg
+
+    @contextmanager
+    @abstractmethod
+    def open(self, item: Any) -> IO:
```

Comment:
Likely needs to become

```suggestion
    def open(self, item: PurePosixPath) -> IO:
```

to set concrete expectations. The `archivist`/`datalad-archives` use case will always come with such paths. I am not sure of particular archive formats use non-POSIX paths internally, but knowing the type of input path should help make any transformations that are necessary.

Are there any archives that use non-path item/member identification schemes?